dataline-workers/src/test/java/io/dataline/workers/singer/SingerDiscoverSchemaWorkerTest.java

```diff
@@ -24,90 +24,97 @@
 
 package io.dataline.workers.singer;
 
-import static io.dataline.workers.JobStatus.FAILED;
-import static io.dataline.workers.JobStatus.SUCCESSFUL;
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.Assert.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import io.dataline.commons.io.IOs;
 import io.dataline.commons.json.Jsons;
+import io.dataline.commons.resources.Resourzes;
 import io.dataline.config.StandardDiscoverSchemaInput;
 import io.dataline.config.StandardDiscoverSchemaOutput;
-import io.dataline.integrations.Integrations;
-import io.dataline.workers.BaseWorkerTestCase;
 import io.dataline.workers.InvalidCredentialsException;
+import io.dataline.workers.JobStatus;
 import io.dataline.workers.OutputAndStatus;
-import io.dataline.workers.PostgreSQLContainerTestHelper;
+import io.dataline.workers.process.ProcessBuilderFactory;
 import java.io.IOException;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.concurrent.TimeUnit;
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.testcontainers.containers.PostgreSQLContainer;
-import org.testcontainers.utility.MountableFile;
 
-public class SingerDiscoverSchemaWorkerTest extends BaseWorkerTestCase {
+public class SingerDiscoverSchemaWorkerTest {
 
-  PostgreSQLContainer db;
+  private static final String IMAGE_NAME = "selfie:latest";
+  private static final JsonNode CREDS = Jsons.jsonNode(ImmutableMap.builder().put("apiKey", "123").build());
 
-  @BeforeAll
-  public void initDb() throws IOException, InterruptedException {
-    db = new PostgreSQLContainer();
-    db.start();
-    PostgreSQLContainerTestHelper.runSqlScript(
-        MountableFile.forClasspathResource("simple_postgres_init.sql"), db);
+  private Path jobRoot;
+  private ProcessBuilderFactory pbf;
+  private ProcessBuilder processBuilder;
+  private Process process;
+  private StandardDiscoverSchemaInput input;
+
+  @BeforeEach
+  public void setup() throws IOException, InterruptedException {
+    jobRoot = Files.createTempDirectory("");
+    pbf = mock(ProcessBuilderFactory.class);
```

Comment:
you can could Mockito. RETURNS_DEEP_STUBS so you don't need to mock all the intermediate objects.

Not sure if it is applicable here.

Reply:
good suggestion, but after looking some i couldn't find anything here that made sense to me given the verifies that i wanted to do.